src/graph/test/DataTest.cpp

```diff
@@ -0,0 +1,164 @@
+/* Copyright (c) 2019 vesoft inc. All rights reserved.
+ *
+ * This source code is licensed under Apache 2.0 License,
+ * attached with Common Clause Condition 1.0, found in the LICENSES directory.
+ */
+
+#include "base/Base.h"
+#include "graph/test/TestEnv.h"
+#include "graph/test/TestBase.h"
+#include "meta/test/TestUtils.h"
+
+DECLARE_int32(load_data_interval_secs);
+
+namespace nebula {
+namespace graph {
+
+class DataTest : public TestBase {
+protected:
+    void SetUp() override {
+        TestBase::SetUp();
+        // ...
+    }
+
+    void TearDown() override {
+        // ...
+        TestBase::TearDown();
+    }
+
+    static void SetUpTestCase() {
+        client_ = gEnv->getClient();
+
+        ASSERT_NE(nullptr, client_);
+
+        ASSERT_TRUE(prepareSchema());
+    }
+
+    static void TearDownTestCase() {
+        ASSERT_TRUE(removeData());
+        client_.reset();
+    }
+
+protected:
+    static AssertionResult prepareSchema();
+
+    static AssertionResult removeData();
+
+    static std::unique_ptr<GraphClient>         client_;
+};
+
+std::unique_ptr<GraphClient>         DataTest::client_{nullptr};
+
+AssertionResult DataTest::prepareSchema() {
+    {
+        cpp2::ExecutionResponse resp;
+        std::string host = folly::stringPrintf("127.0.0.1:%u", gEnv->storageServerPort());
+        std::string cmd = "ADD HOSTS " + host;
+        auto code = client_->execute(cmd, resp);
+        if (cpp2::ErrorCode::SUCCEEDED != code) {
+            return TestError() << "Do cmd:" << cmd << " failed";
+        }
+        meta::TestUtils::registerHB(network::NetworkUtils::toHosts(host).value());
+    }
+    {
+        cpp2::ExecutionResponse resp;
+        std::string cmd = "CREATE SPACE mySpace(partition_num=1, replica_factor=1)";
+        auto code = client_->execute(cmd, resp);
+        if (cpp2::ErrorCode::SUCCEEDED != code) {
+            return TestError() << "Do cmd:" << cmd << " failed";
+        }
+    }
+    sleep(FLAGS_load_data_interval_secs + 1);
+    {
+        cpp2::ExecutionResponse resp;
+        std::string cmd = "USE mySpace";
+        auto code = client_->execute(cmd, resp);
+        if (cpp2::ErrorCode::SUCCEEDED != code) {
+            return TestError() << "Do cmd:" << cmd << " failed";
+        }
+    }
+    {
+        cpp2::ExecutionResponse resp;
+        std::string cmd = "CREATE TAG man()";
+        auto code = client_->execute(cmd, resp);
+        if (cpp2::ErrorCode::SUCCEEDED != code) {
+            return TestError() << "Do cmd:" << cmd << " failed";
+        }
+    }
+    {
+        cpp2::ExecutionResponse resp;
+        std::string cmd = "CREATE TAG company(name string)";
+        auto code = client_->execute(cmd, resp);
+        if (cpp2::ErrorCode::SUCCEEDED != code) {
+            return TestError() << "Do cmd:" << cmd << " failed";
+        }
+    }
+    {
+        cpp2::ExecutionResponse resp;
+        std::string cmd = "CREATE EDGE work()";
```

Comment:
add some cases:
1. create edge e1(); create edge e2(); 
2. create tag t1(); create e1();
3. a very very long sentence /* or repeatly add 1000 times  */

Reply:
@whitewum  👍
> 3\. a very very long sentence /* or repeatly add 1000 times  */

You inspired me, I have a few small questions:

1) Is there a limit to the length of the SQL statement?
2) Is there a limit on the number of tag/edge columns?
3) Is there a limit to the length of each record when inserting vertex/edge?

Reply:
Good! Add tests to cover this limits (and will not be changed by others in the future.)